components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/IsMemberOfAnyOfGroupsFunctionImpl.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.conditional.auth.functions.user;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.exception.FrameworkException;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Function to check whether the specified user belongs to one of the groups specified in the list of user groups.
+ */
+public class IsMemberOfAnyOfGroupsFunctionImpl implements IsMemberOfAnyOfGroupsFunction {
+
+    private static final Log LOG = LogFactory.getLog(IsMemberOfAnyOfGroupsFunctionImpl.class);
+
+    @Override
+    public boolean isMemberOfAnyOfGroups(JsAuthenticatedUser user, List<String> groupNames) {
+
+        boolean result = false;
+        String tenantDomain = user.getWrapped().getTenantDomain();
+        String userStoreDomain = user.getWrapped().getUserStoreDomain();
+        String username = user.getWrapped().getUserName();
+
+        // Build the user store domain aware role name list.
+        List<String> groupsWithDomain = getDomainAwareGroupNames(userStoreDomain, groupNames);
+        try {
+            UserRealm userRealm = Utils.getUserRealm(user.getWrapped().getTenantDomain());
+            if (userRealm != null) {
+                UserStoreManager userStore = Utils.getUserStoreManager(tenantDomain, userRealm, userStoreDomain);
+                if (userStore != null) {
+                    // List returned by the user store will contain the roles and groups.
+                    String[] roleListOfUser = userStore.getRoleListOfUser(username);
+                    result = Arrays.stream(roleListOfUser).anyMatch(groupsWithDomain::contains);
```

Comment:
Shouldn't we filter out the roles and only match with the groups here?

Reply:
since we are prepending userstore domain names to the given roles list, we don't need to specifically filter the roles

Reply:
What about the PRIMARY userstore?
will the behaviour be the same?

Reply:
For the primary userstore, the roles will have their domain(internal,application etc) but the groups will not have the domain name.
In [1], if the domain is primary, we do not prepend userstore domain name.

[1] - https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/62/files#diff-885c209dcfc8d8d652f0826c1d8b546410cf4f542ced6066d4eb0f55f41bb312R82